bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -330,6 +331,28 @@ private BigtableOptions resolveLegacyOptions(BigtableOptions options) throws IOE
     return options;
   }
 
+
+  /**
+   * Return options with legacy input options, if any, resolved into currently supported options.
+   */
+  public synchronized BigtableClusterName getClusterName() throws IOException {
+    if (options.getClusterName() == null) {
+      try (BigtableClusterUtilities util =
+          BigtableClusterUtilities.forInstance(options.getProjectId(), options.getInstanceId())) {
+        final ListClustersResponse clusters = util.getClusters();
+        Preconditions.checkState(clusters.getClustersCount() == 0,
+          "Project '%s' / Instance '%s' has %d clusters. There must be exactly 1 for this operation to work.",
+          options.getProjectId(), options.getInstanceId(), clusters.getClustersCount());
+        String clusterName = util.getSingleCluster().getName();
```

Comment:
I thought getSingleCluster() does all the checking already. Why do we need to getClusters() first?